packages/inter-protocol/src/reserve/assetReserveKit.js

```
@@ -166,6 +198,67 @@ export const prepareAssetReserveKit = async (
           facets.helper.writeMetrics();
         },
       },
+      withdrawalFacet: {
+        Withdraw() {
+          /** @type {OfferHandler<Promise<string>>} */
+          const handler = async seat => {
+            try {
+              guard.assert();
+              const { state } = this;
+              const { helper } = this.facets;
+              const { want } = seat.getProposal();
+              const withdrawMap = new Map();
+              const depositMap = new Map();
+
+              const wantEntries = Object.entries(want);
+              for (const [keyword, amount] of wantEntries) {
+                const amountOut = AmountMath.coerce(amount.brand, amount);
+                const collateralKeyword = helper.getKeywordForBrand(
+                  amountOut.brand,
+                );
+                let collateralOut = amountOut;
+                const collateralSoFar = withdrawMap.get(collateralKeyword);
+                if (collateralSoFar) {
+                  collateralOut = AmountMath.add(collateralSoFar, amountOut);
+                }
+                withdrawMap.set(collateralKeyword, collateralOut);
+                depositMap.set(keyword, amountOut);
+              }
+
+              const withdrawRecord = Object.fromEntries(withdrawMap.entries());
+              const depositRecord = Object.fromEntries(depositMap.entries());
+              atomicTransfer(
+                zcf,
+                state.collateralSeat,
+                seat,
+                withdrawRecord,
+                depositRecord,
+              );
+
+              guard.assert();
```

ugh. this is a possible failure after the commit point.

I suggest an explicit `// COMMIT` comment at the commit point before `atomicTransfer`.

And add a comment justifying why we do not **abort the vat** per [death before confusion](https://github.com/Agoric/agoric-sdk/wiki/Prepare-Commit).

by way of example:
https://github.com/Agoric/agoric-sdk/blob/51b6790bbb491ecc62603d8333548071a8323470/packages/fast-usdc-contract/src/exos/liquidity-pool.ts#L159-L161

... which reminds me that even `atomicTransfer` / `atomicRearrange` can fail, and it's after the commit point too.

In fast-usdc, we didn't put a `try / catch` around it; we just let it propagate to wherever unhandled exceptions go. I think we considered red-alert logging but opted not to; I don't recall clearly why... #10659 , linked from #10684 , has some info.

---

Thanks, these are good simplifications that make the code much more understandable.